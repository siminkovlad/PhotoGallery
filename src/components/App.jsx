import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Container } from 'react-bootstrap';

import {
    Header,
    Main,
    Footer,
    Loader
} from './index';
import { GlobalStyle } from '../styles/globalStyles';

import { requestPhotos } from '../store/fetchPhotos';

export const App = () => {
    const loading = useSelector(state => state.fetchPhotos.loading);
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(requestPhotos());
    }, []);

    return (
        <Container fluid>
            <GlobalStyle />
            <Header />
            {loading ? <Loader /> : <Main />}
            <Footer />
        </Container>
    );
};
