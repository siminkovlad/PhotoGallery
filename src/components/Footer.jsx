import React from 'react';
import { Col, Row } from 'react-bootstrap';

import { FooterStyled } from '../styles/styledFooter';

export const Footer = () => (
    <Row>
        <Col>
            <FooterStyled>Â© {new Date().getFullYear()}</FooterStyled>
        </Col>
    </Row>
);
