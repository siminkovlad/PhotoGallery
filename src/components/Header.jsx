import React from 'react';
import { Col, Row } from 'react-bootstrap';

import { HeaderStyled } from '../styles/styledHeader';

export const Header = () => (
    <Row>
        <Col>
            <HeaderStyled>
                <h1>PHOTO GALLERY</h1>
            </HeaderStyled>
        </Col>
    </Row>
);
