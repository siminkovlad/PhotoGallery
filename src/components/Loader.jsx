import React from 'react';

import { LoaderStyled, DivStyled } from '../styles/styledLoader';

export const Loader = () => (
    <LoaderStyled>
        <DivStyled />
    </LoaderStyled>
);
