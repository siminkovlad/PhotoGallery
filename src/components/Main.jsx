import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Route, Link, useHistory } from 'react-router-dom';
import {
    Row,
    Image,
    Modal,
    Form,
    Button,
    Carousel
} from 'react-bootstrap';

import {
    hidePopUp,
    requestComments,
    showPopUp
} from '../store/fetchComments';
import { postComment } from '../store/postComment';

export const Main = () => {
    const [id, setId] = useState(null);
    const fetchedPhotos = useSelector(state => state.fetchPhotos.fetchedPhotos);
    const { show, comments } = useSelector(state => state.fetchComments);
    const dispatch = useDispatch();
    const history = useHistory();

    useEffect(() => {
        const searchParams = window.location.pathname.slice(1);

        if (searchParams) openModalHandler();
    }, []);

    const openModalHandler = async () => {
        await dispatch(showPopUp());
        setId(window.location.pathname.slice(1));
        dispatch(requestComments());
    };

    const sendRequest = e => {
        e.preventDefault();

        dispatch(postComment());
    };

    const onCloseModalHandler = () => {
        history.push('/');
        dispatch(hidePopUp());
    };

    return (
        <Row className="main">
            <Carousel interval={5000}>
                {fetchedPhotos.length && fetchedPhotos.map(item => (
                        <Carousel.Item key={item.id}>
                            <Link to={`/${item.id}`}>
                                <Image src={item.url} onClick={openModalHandler} />
                            </Link>
                        </Carousel.Item>
                    )
                )}
            </Carousel>
            <Route path="/:userSearch">
                {comments && +id === comments.id &&
                <Modal show={show} onHide={onCloseModalHandler} animation={false} size="xl">
                    <Modal.Header closeButton />
                    <Modal.Body>
                        <Image className="pop-up-div pop-up-img" src={comments.url} fluid />
                        <div className="pop-up-div">
                            <h1>Comments:</h1>
                            <ul>
                                {comments?.comments.map(item => (
                                    <li key={item.id}>{item.text}</li>
                                ))}
                                {!comments.comments.length && <div>No comments yet :( You can be first!</div>}
                            </ul>
                        </div>
                    </Modal.Body>
                    <Modal.Footer>
                        <Form id="comments" onSubmit={e => sendRequest(e)}>
                            <Form.Group controlId="formBasicName">
                                <Form.Label>Your name:</Form.Label>
                                <Form.Control type="input" placeholder="Enter your name" />
                            </Form.Group>
                            <Form.Group controlId="formBasicComment">
                                <Form.Label>Your comment:</Form.Label>
                                <Form.Control type="input" placeholder="Enter your comment" />
                            </Form.Group>
                            <Button variant="primary" type="submit">Comment</Button>
                        </Form>
                    </Modal.Footer>
                </Modal>}
            </Route>
        </Row>
    );
};
