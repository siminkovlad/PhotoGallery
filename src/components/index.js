export { Footer } from './Footer';
export { Main } from './Main';
export { Header } from './Header';
export { Loader } from './Loader';
export { App } from './App';
