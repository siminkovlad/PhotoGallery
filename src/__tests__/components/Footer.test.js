import React from 'react';
import toJson from 'enzyme-to-json';

import { Footer } from '../../components/Footer';

describe('Footer tests', () => {
    const footerContainer = shallow(<Footer/>);

    it('Create snapshot', () => {
        expect(toJson(footerContainer)).toMatchSnapshot();
    });

    it('Renders text in footer', () => {
        expect(footerContainer.text(`Â© ${new Date().getFullYear()}`));
    });
});
