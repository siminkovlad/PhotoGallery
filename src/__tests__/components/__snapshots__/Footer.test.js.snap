// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests Create snapshot 1`] = `
<Row
  noGutters={false}
>
  <Col>
    <styled.footer>
      Â© 
      2021
    </styled.footer>
  </Col>
</Row>
`;
